neware_parser/DataEngine.py

```diff
@@ -84,6 +84,7 @@ def compute_scale(
         PickleDump.scale(filename, patches, keys, scales, cycles)
 
```

Comment:
A nice protocol is a class which takes some info (like a path_to_plot_data) and has two methods.
One of the methods takes in a big mapping of barcodes to computed results or just data and it dumps it to the right files.
The other method looks into the path and loads all the data in the same format that it was put in. For instance, if each barcode produces a file and different iteration ids have different files, then it loads the data into a dictionary with a key being the iteration id and the other key being the barcode.